### DISCUSSION TITLE
Read HEIC image - output to jpg

### DISCUSSION THREAD

**Discussion by danielsetreus** [ImageMagick/ImageMagick#5017] [Help] [answered]
I've seen alot of questions regarding reading HEIC images, and ImageMagick support for it. I really don't want to start a new one just for the sake of it, but i've yet to find a non closed thread or issue that fits. So here it goes: 

```
> magick -version
Version: ImageMagick 7.1.0-29 Q16-HDRI x86_64 19841 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): heic jng jpeg png zlib
Compiler: gcc (9.4)
```

```
> magick -list format
...
HEIC* r--   High Efficiency Image Format (1.4.0)
HEIF* r--   High Efficiency Image Format (1.4.0)
...
```

```
> convert -list configure | grep -i "delegates"
DELEGATES             heic jng jpeg png zlib
DELEGATES             heic jng jpeg png zlib
```

```
> cat /etc/*-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.4 LTS"
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

``` 
> magick test.heif test.jpg
magick: no images for write '-write' 'test.jpg' at CLI arg 2 @ error/operation.c/CLINoImageOperator/4919.
```

It seems that even though ImageMagick claims it will be able to read heic/heif in actuality it cannot. libheif is also installed from source (https://github.com/strukturag/libheif/releases). I've tried with both the most recent version 1.12.0, and (as seen above) 1.4.0 - since that was used in an example that claimed they had it working. 

I'd appreciate any assistance. Thanks!

**Comment by snibgo** [comment]
Please link to your test.heif.

**Reply by danielsetreus** [reply]
https://easyupload.io/gbiwux - valid for 30 days from now

**Comment by snibgo** [comment]
Cute dog.

With IM v7.1.0-4, pre-built on Windows 8.1, it converts with no problem. It contains an embedded profile. If you want sRGB, I suggest converting like this:
```
magick test.heif -profile sRGB.icc x.jpg
```

Your delegate list is small:
```
Delegates (built-in): heic jng jpeg png zlib
```
Compare that to the list I get:
```
f:\web\im>%IMG7%magick -version
Version: ImageMagick 7.1.0-4 Q16 x64 2021-07-18 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Visual C++: 192930038
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms
 lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
```
I suspect that "-profile" need lcms. Perhaps you also need some other built-in delegate to successfully read heif files.

**Comment by Jmuccigr** [comment]
FWIW I get the same error, though I can work with .heic images off my iPhone with no problem. Working on a Catalina Mac.

Version: ImageMagick 7.1.0-29 Q16-HDRI x86_64 19841 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib
Compiler: gcc (4.2)

**Comment by danielsetreus** [comment, marked as answer]
I suspect that there might have been something wrong with the image file I had (sent over Slack so they may have destroyed the file somehow). I was able to follow, or be inspired by, the answer at this link https://askubuntu.com/questions/1042436/how-to-install-delegate-libraries-for-image-magick-7-0-7

Also installed `libheic-dev`. 

Basically - install all the dependencies of the pre-built version of imagemagick, compile IM confired `--with-heic=yes`. 

After that `magick sample1.heic sample1.jpg` actually worked - using the HEIC samples downloadable here https://filesamples.com/formats/heic

As someone new to compiling things from source I have to say that the documentation on how to do this with ImageMagick is somewhat lacking. They could at least mention that there are *a lot* of dependencies you'll want in order to actually make ImageMagick useful.

**Comment by MilesSowden** [comment]
Similar problem as danielsetreus. Ubuntu is grabing version 6. 
Solved it by (horrors) installing the windows version and using the DOS command line :-(
